### DISCUSSION TITLE
Alias import paths using glob

### DISCUSSION THREAD

**Discussion by jboler** [parcel-bundler/parcel#8836] [Q&A] [answered]
I'd like to be able to do:
```ts
import { method } from '#/workspace/foo.ts';
```
Instead of:
```ts
import { method } from '/src/workspace/foo.ts';
```
(The absolute path works totally fine but I'm migrating a CRA 3000+ file codebase that's using a tsconfig path `"paths": { "#/*": ["src/*"] }` that I'd like to keep.

From the docs I'd expect to be able to use a glob alias like this:
```ts
  "alias": {
    "#/*": "./src/$1"
  },
```
I've also tried:
```ts
  "alias": {
    "#/*": "/src/$1"
  },
```
But whatever I try it doesn't work and I get `TypeError: (0 , _fooTs.method) is not a function`

My directory structure:
```
- package.json
- .parcelrc
- src/index.ts
- src/workspace/foo.ts
```

My `.parcelrc`:
```
{
  "extends": "@parcel/config-default",
  "resolvers": [
    "@parcel/resolver-glob",
    "..."
  ]
}
```

Any help would be appreciated.

**Comment by jboler** [comment, marked as answer]
I figured it out. `#` isn't supported as a prefix and to match multiple path segments the glob has to be `**`:
```json
  "alias": {
    "_/**": "/src/$1"
  },
```

**Reply by mischnic** [reply]
1. That `#` handling is definitely not correct: https://github.com/parcel-bundler/parcel/issues/8839
2. (You don't need the glob resolver for this glob-in-alias feature. )

**Reply by jboler** [reply]
For anyone else looking for this solution, to support import paths like this:
```ts
import { method } from '@/workspace/foo.ts';
```
use a parcel alias like this:
```json
  "alias": {
    "@/**/*": "/src/$1/$2"
  },
```
